xcube/core/gen2/generator.py

```diff
@@ -0,0 +1,128 @@
+# The MIT License (MIT)
+# Copyright (c) 2021 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from typing import Optional
+
+import xarray as xr
+
+from xcube.core.store import DataStorePool
+from xcube.util.assertions import assert_instance
+from xcube.util.progress import observe_progress
+from .combiner import CubesCombiner
+from .config import CubeGeneratorConfig
+from .opener import CubesOpener
+from .progress import ApiProgressCallbackObserver
+from .progress import ConsoleProgressObserver
+from .writer import CubeWriter
+
+
+class CubeGenerator:
+    """
+    Generator tool for data cubes.
+
+    Creates cube views from one or more cube stores, resamples them to a
+    common grid, optionally performs some cube transformation, and writes
+    the resulting cube to some target cube store.
+
+    :param gen_config: Cube generation configuration.
+    :param store_pool: An optional pool of pre-configured data stores
+        referenced from *gen_config* input/output configurations.
+    :param verbose: Whether to output progress information to stdout.
+    """
+
+    def __init__(self,
+                 gen_config: CubeGeneratorConfig,
+                 store_pool: DataStorePool = None,
+                 verbose: bool = False):
+        assert_instance(gen_config, CubeGeneratorConfig, 'gen_config')
+        if store_pool is not None:
+            assert_instance(store_pool, DataStorePool, 'store_pool')
+
+        self._gen_config = gen_config
+        self._store_pool = store_pool if store_pool is not None \
+            else DataStorePool()
+        self._verbose = verbose
+
+    @classmethod
+    def from_file(cls,
```

Comment:
As you can pass more than one file as parameter: Rename to from_files or, what I prefer, from_config

Reply:
No, it is the one file positional arg, the other file params are keywork arguments, hence options.

Reply:
In this case the first sentence of the docstring is a bit misleading. I still prefer "config", even more so as the configuration may be piped.

Reply:
`form_config` doesn't make sense, because it is always constructed from configuration. The difference here is that configuration comes from a _file_ rather than a string, pipe stream, HTTP, whatsoever.